bin/export-codePost-grades

```diff
@@ -0,0 +1,254 @@
+    _print_info("Command line parameters: {}".format(outparams))
+
+    # Load YAML configuration file
+
+    # Resolve possible locations for this configuration file
+    # FIXME: This is hard-coded, whereas it should be a (documented) constant.
+    possible_locations = ["codepost-config.yaml", ".codepost-config.yaml",
+                          "~/codepost-config.yaml", "~/.codepost-config.yaml"]
+
+    location = None
+    for p in possible_locations:
+        path = os.path.abspath(os.path.expanduser(p))
+        if os.path.exists(path):
+            location = path
+            break
+
+    if location == None:
+        _print_err(
+            "No codepost-config.yaml configuration file detected. Exiting.",
+            fatal=8)
+    else:
+        _print_info("Configuration path: {}".format(location))
+
+    config = load(open(location), Loader=Loader)
+    _print_info("Configuration content: {}".format(config))
+
+    return params, config
+
+
+clparams, config = process_command_line(parser)
+
+grades = cP.get_course_grades(
+    api_key=config["api_key"],
+    course_name=config["course_name"],
+    course_period=config["course_period"])
+
+# JSON output
+if clparams["json"]:
+    import json
+    if clparams["pretty"]:
+        print(json.dumps(grades, indent=2))
+    else:
+        print(json.dumps(grades))
+    sys.exit(0)
+
+
+course = cP.get_course_roster_by_name(
+    api_key=config["api_key"],
+    course_name=config["course_name"],
+    course_period=config["course_period"])
+
+# CSV output
+
+# Figure out headers
+headers = clparams.get("a", list())
+if headers == None or len(headers) == 0:
+    headers = []
+    for aid in course["assignments"]:
+        assignment_info = cP.get_assignment_info_by_id(
+            api_key=config["api_key"],
+            assignment_id=aid)
+        headers.append(assignment_info["name"])
+
+    if not clparams["show_empty"]:
+        used_assignments = set()
+        for student_grades in grades.values():
+            used_assignments = used_assignments.union(
+                set(student_grades.keys()))
+
+        headers = [header for header in headers if header in used_assignments]
+
+# Figure out students
+students = clparams.get("s", list())
+if len(students) > 0:
+    # Normalize input
+    students = map(lambda n: config["user_pattern"].format(n), students)
+
+else:
+    students = course.get("students", list())
+
+    if clparams["include_inactive"]:
+        students += course.get("inactive_students", list())
+
+csv_lines = []
+
+header_line = None
+
+if clparams["blackboard"]:
+    if not "lms_ids" in config:
+        _print_warn(
+            "Blackboard column IDs requested but not available in configuration")
+    else:
+        lms_ids = config["lms_ids"]
+        lms_format = config.get("lms_id", '"{name} | {id}"')
+
+        def format_header_title(title):
+            lms_id = lms_ids.get(title, "")
+            return lms_format.format(
```

Comment:
Oh wow! This is the code snippet I was looking for but didn't quite know how to do. This is exactly why I don't demo the output "--blackboard" flag in the README.md, because then it required I configured *all* the columns, because it was ugly for those that were missing. This is a much better way to do it. Thank you @rfreling!!